Salesforce/Transformer/Plugins/DateTimeTransformer.php

```diff
@@ -71,6 +71,12 @@ public function transformInbound(TransformerPayload $payload)
             $type = Type::getType($type);
         }
 
+        $dateTime = new \DateTime($value);
```

Comment:
If the doctrine mapping uses an immutable type, `\DateTime' will throw an error.

```suggestion
        $dateTime = false != preg_match('/_immutable$/', $type->getName()) ? new \DateTimeImmutable($value) : new \DateTime($value);
```